### ISSUE TITLE
Zend inconsistency in the Reflection API for methods coming from traits

### ISSUE THREAD

**Issue by stof** [facebook/hhvm#4702]
While trying to identify an issue with some PHP code, I identified a case where HHVM differs from PHP 5.5: http://3v4l.org/K2nbU

**Comment by paulbiss** [comment]
Hmm, looks like we match older versions of php 5 and newer revs of php 7.

**Comment by jwatzman** [comment]
Yeah, we match all the php7 versions except one random one. So looks to me like PHP itself hasn't thought this through and isn't consistent about it... and we match PHP7... do we care to fix it to match 5.6? (Is "fix" the right word, even? :))

**Comment by stof** [comment]
I would be perfectly fine if you consider it as a won't fix actually. It is kind of an edge case anyway.

I detected it by chance while debugging an infinite loop in Behat 2.5 on PHP 5.5 and 5.6, and I thought it could be worth reporting it to raise awareness about this case.

**Comment by jwatzman** [comment]
Yeah, definitely worth reporting, even if we end up wontfix'ing it. I'm gonna mark it low pri but keep open -- when someone gets time, it might be worth digging into what the difference is and if PHP's behavior is deliberate, and if we should change something (or not) and/or report a bug to PHP5/PHP7.

**Comment by paulbiss** [comment]
This is also going to be inconsistent in repo mode for us so we should probably fix it. It's also now consistent in PHP 7 and 7.1. The fix is probably just to not return traits from here:

https://github.com/facebook/hhvm/blob/c6b96eab21be48f88d19241aab5b78b482aacb05/hphp/runtime/ext/reflection/ext_reflection.cpp#L901-L918

**Comment by lexidor** [comment]
![image](https://user-images.githubusercontent.com/31805625/82456508-9bfc1080-9ab4-11ea-8ab3-aa3152244360.png)

Hhvm 4.56 matches php 5.4.0 -> 5.4.10.
